src/emailsearchjob.cpp

```diff
@@ -41,8 +41,9 @@ bool EmailSearchJob::firstRun()
 {
     KSharedConfigPtr config = KSharedConfig::openConfig("kdenowrc");
     KConfigGroup generalGroup(config, "General");
-    QString uidName = generalGroup.readEntry("UIDNEXT", QString());
-    if ("FooBar" == uidName) {
+    QString state = generalGroup.readEntry("STATE", QString());
```

Comment:
KConfigGroup.readEntry will return QString() if it doesn't find any entry "STATE". On first run of the application, it won't have "STATE" entry. So, if it returns QString(), then next if statement would be evaluated to false on first run.